score-client/src/main/java/bio/overture/score/client/config/AzureConfig.java

```diff
@@ -19,24 +19,21 @@
 
 import bio.overture.score.client.upload.UploadService;
 import bio.overture.score.client.upload.azure.AzureUploadService;
+import bio.overture.score.client.upload.s3.S3UploadService;
 import bio.overture.score.client.util.AzurePresignedUrlValidator;
 import bio.overture.score.client.util.PresignedUrlValidator;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Profile;
 
+// Note: Name the beans as profile-name+class-type to be able to dynamically fetch beans at runtime based on score-server profiles
 @Configuration
-@Profile("azure")
 public class AzureConfig {
 
   @Bean
-  public UploadService uploadService() {
+  public AzureUploadService azureUploadService(){
```

Comment:
Why do we need to provide the exact named class instead of an implementation of the parent class (`UploadService`). The note says it has to be done but I don't really understand why, what's the reason/justification?